src/components/ConfigEditor/AzureCredentialsForm.tsx

```diff
@@ -49,23 +64,27 @@ export const AzureCredentialsForm: FunctionComponent<Props> = (props: Props) =>
     }
 
     return opts;
-  }, [props.userIdentityEnabled, props.managedIdentityEnabled, props.oboEnabled]);
+  }, [userIdentityEnabled, managedIdentityEnabled, workloadIdentityEnabled, props.oboEnabled]);
 
   const onAuthTypeChange = (selected: SelectableValue<AzureAuthType>) => {
+    const defaultAuthType = userIdentityEnabled
```

Comment:
I think this code would be clearer if using switches or ifs. I'm getting confused looking at it :laughing:

Reply:
Don't worry, this is all going to be refactored soon 😉